app/graphql/resolvers/create_user.rb

```diff
@@ -19,17 +24,28 @@ def call(_obj, args, ctx)
         "Email taken by %s %s." % [email_user.first_name, email_user.last_name]
       )
     end
-
+    
     @new_user = User.new(
       first_name: args[:first_name],
       last_name: args[:last_name],
       email: args[:email],
       password: args[:password],
       password_confirmation: args[:password_confirmation],
       created_at: Time.now,
+      profile_picture: args["attachment"] || nil,
     )
+    if args["role"] == "Contributor"
```

Comment:
Instead of hard-coding the types of roles, ideally you should use role slugs instead. So something like @new_user.roles << Role.find_by(slug: args["role"]) if args["role"]

However, users can and do have more than one role, so really role should be [String!]. You can loop over the roles and add each one (probably should check for invalid slugs which my code doesn't do).

Also, roles should only be allowed to be added by admins. Right now this is sort of a security hole. If roles is nil, do nothing, otherwise check auth status and if it checks out run the loop code.

Reply:
Role String will be added later, will create an issue